tests/internal/testutils/templates.go

```diff
@@ -42,3 +43,68 @@ func FindAllTemplates(t *testing.T, templateUrl string) []TemplateInfo {
 	}
 	return infos
 }
+
+// UpdateOptions returns the set of integration.ProgramTestOptions that should be applied for the
+// given template.
+func UpdateOptions(templateName string) integration.ProgramTestOptions {
+
+	// For our core Getting Started templates, we test the full end to end experience,
+	// ensuring updates succeed and subsequent operations produce no changes.
+	skipFullTest := !isGettingStartedTemplate(templateName)
+
+	return integration.ProgramTestOptions{
+		// Skip running a full update.
+		SkipUpdate: skipFullTest,
+		// Skip running a refresh after the update, expecting no changes.
+		SkipRefresh: skipFullTest,
+		// Skip running a preview after the update, expecting no changes.
+		SkipPreview: skipFullTest,
+		// Skip running a stack export and import after the update.
+		SkipExportImport: skipFullTest,
+	}
+}
+
+// isGettingStartedTemplate returns whether the specified template is one that is used in
+// our Getting Started flows.
+func isGettingStartedTemplate(templateName string) bool {
```

Comment:
A potential alternative to maintaining the list in this function would be to base it on whether the template has `important: true`, which is the subset of templates shown by `pulumi new` by default, and also appears to currently align with the getting started list you have here. You can get at this information via `templateInfo.Template.Important`.

Reply:
Oh, that's much nicer. I didn't realize that was there. Thanks for the tip!